lib/auth/v4/headerAuthCheck.ts

```diff
@@ -26,7 +27,7 @@ export function check(
     log: Logger,
     data: { [key: string]: string },
     awsService: string
-) {
+): { err: null | ArsenalError | Error, params?: AuthV4RequestParams } {
```

Comment:
It seems this cannot return an `Error`.

The case would be a failure to utf8.encode:
```
    const stringToSign = constructStringToSign({ ... });
    if (stringToSign instanceof Error) {
        return { err: stringToSign };
    }
```

But according to the docs this function does not return an error, it throws on error:

> Encodes any given JavaScript string (string) as UTF-8, and returns the UTF-8-encoded version of the string. It throws an error if the input string contains a non-scalar value, i.e. a lone surrogate. (If you need to be able to encode non-scalar values as well, use WTF-8 instead.)